src/app/clusters/ota-requestor/DefaultOTARequestor.h

```diff
@@ -111,6 +111,41 @@ class DefaultOTARequestor : public OTARequestorInterface, public BDXDownloader::
      */
     CHIP_ERROR Init(Server & server, OTARequestorStorage & storage, OTARequestorDriver & driver, BDXDownloader & downloader);
 
+    /**
+     * Called to override the default device config vendor ID value
+     */
+    void SetQueryImageVendorId(VendorId vendorId)
+    {
+        // Vendor ID 0x0000 is a reserved value
+        if (vendorId != 0)
+        {
+            mVendorId.SetValue(vendorId);
+        }
+    }
+
+    /**
+     * Called to override the default device config product ID value
+     */
+    void SetQueryImageProductId(uint16_t productId)
+    {
+        // Product ID 0x0000 is a reserved value
+        if (productId != 0)
+        {
+            mProductId.SetValue(productId);
+        }
+    }
+
+    /**
+     * Called to override the default device config hardware version value
+     */
+    void SetQueryImageHardwareVersion(uint16_t hwVersion)
+    {
+        if (hwVersion != 0)
```

Comment:
Is 0 disallowed?

Reply:
No it is not. hwVersion == 0 needs to be allowed

Reply:
I'll explicitly check in main.cpp instead then so this check can be removed.